test/services/database/user_comment_repository_test.dart

```diff
@@ -0,0 +1,174 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/models/database/user_comment/user_comment_data.dart";
+import "package:proxima/models/database/user_comment/user_comment_firestore.dart";
+import "package:proxima/services/database/firestore_service.dart";
+import "package:proxima/services/database/user_comment_repository_service.dart";
+
+import "../../mocks/data/firestore_user.dart";
+import "../../mocks/data/firestore_user_comment.dart";
+import "../../mocks/data/user_comment_data.dart";
+
+void main() {
+  group("Testing user comments repository", () {
+    late FakeFirebaseFirestore fakeFirestore;
+    late UserCommentReposittoryService userCommentRepository;
+
+    late UserFirestore user;
+    late UserCommentFirestoreGenerator userCommentGenerator;
+    late UserCommentDataGenerator userCommentDataGenerator;
+    late CollectionReference<Map<String, dynamic>> userCommentsCollection;
+
+    setUp(() async {
+      fakeFirestore = FakeFirebaseFirestore();
+
+      final container = ProviderContainer(
+        overrides: [
+          firestoreProvider.overrideWithValue(fakeFirestore),
+        ],
+      );
+
+      userCommentRepository =
+          container.read(userCommentRepositoryServiceProvider);
+
+      user = (await FirestoreUserGenerator.addUsers(fakeFirestore, 1)).first;
+
+      userCommentGenerator = UserCommentFirestoreGenerator();
+      userCommentDataGenerator = UserCommentDataGenerator();
+      userCommentsCollection = fakeFirestore
+          .collection(UserFirestore.collectionName)
+          .doc(user.uid.value)
+          .collection(UserCommentFirestore.userCommentSubCollectionName);
+    });
+
+    group("getting user comments", () {
+      test("should return an empty list if the user has no comments", () async {
+        final userComments =
+            await userCommentRepository.getUserComments(user.uid);
+
+        expect(userComments, isEmpty);
+      });
+
+      test("should return the user's comments", () async {
+        final userComments = await userCommentGenerator.addComments(
+          3,
+          user.uid,
+          userCommentRepository,
+        );
+
+        final fetchedUserComments =
+            await userCommentRepository.getUserComments(user.uid);
+
+        expect(fetchedUserComments, unorderedEquals(userComments));
+      });
+
+      test("should throw an error if a document has missing fields", () async {
```

Comment:
I just want to point your attention that you tested something very similar (slightly lower level) in `user_comment_data_test.dart`. Are you sure you want to keep this test? Tests are nice, but also complicate code maintanibility. 
I'll leave this to your judgement.

Reply:
I agree with you, this is testing to a bit lower level.
However, I designed that test to cover a part of the code that isn't covered otherwise.
So, do you think it is worth to keep it ?

Reply:
I'm feeling like this is more of a test to add coverage than a real test bringing interest to the codebase. It maybe makes sense to get rid of the other test.
However, I know that you really like tests and coverage; and keeping both tests make sense. I let you do what you think is best.

Reply:
That's good for me, don't worry c: